pep-9999.rst

```diff
@@ -0,0 +1,670 @@
+:::::::::::::::::::::::::
+
+Each entry is a string giving a single classification value for the distribution.
+Classifiers are described in PEP 301.
+
+Examples::
+
+    Classifier: Development Status :: 4 - Beta
+    Classifier: Environment :: Console (Text Based)
+
+Tools SHOULD issue an informational warning if this field contains a
+licensing related Classifier string starting with the `License::` prefix and
+SHOULD suggest the use of a `License Expression` in the `License` field instead.
+
+If the `License` field is present and contains a valid License Expression,
+publishing tools MUST NOT also provide any licensing related Classifiers entries.
+
+However, for compatibility with existing publishing and installation processes,
+licensing related Classifiers entries SHOULD continue to be accepted if the
+License field is absent or does not contain a valid License Expression.
+
+Publishing tools MAY infer a License Expression from the provided Classifiers
+entries if they are able to do so unambiguously.
+
+However, no new licensing related classifiers will be added, with anyone
+requesting them being directed to use a License Expression in the License field
+instead. Note that the licensing related Classifiers may be deprecated in a
+future PEP.
+
+
+Mapping legacy Classifiers to new License expressions
+'''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+Publishing tools MAY infer or suggest an equivalent `License Expression` from
+the provided License or Classifiers information if they are able to do so
+unambiguously. For instance, if a package only has this license classifier::
+
+    Classifier: `License :: OSI Approved :: MIT License`
+
+Then the corresponding value for License using a valid license expression to
+suggest would be::
+
+    License: MIT
+
+
+Here are mappings guidelines for the legacy classifiers:
+
+- Classifier `License :: Other/Proprietary License` becomes License: `Proprietary` expression.
+
+- Classifier `License :: Public Domain` becomes License: `Public-Domain` expression.
+
+- The generic and ambiguous Classifiers `License :: OSI Approved`
+  and `License :: DFSG approved` do not have an equivalent license expression.
+
+- The generic and sometimes ambiguous Classifiers `License :: Free For Educational Use`,
+  `License :: Free For Home Use`, `License :: Free for non-commercial use`,
+  `License :: Freely Distributable`, `License :: Free To Use But Restricted`,
+  and `License :: Freeware` are mapped to the generic License: `Proprietary` expression.
+
+The remainder of the `Classifiers` using a `License::` prefix map to a simple
+single license expression using the corresponding SPDX license identifiers.
+
+
+Summary of Differences From PEP 566
+===================================
+
+* Metadata-Version is now 2.2.
+
+* Added one new field: ``License-File``
+
+* Updated the documentation of two fields:  ``License`` and ``Classifiers``
+
+
+License Expression Library Reference implementation
```

Comment:
opinion: It is important to separate the discussion of the standard from the discussion about the implementation.

This section (and the discussion of a library being developed for licensing) should probably be dropped from the PEP -- it's not relevant to the metadata update and going into too much detail of how-to-do-this, unnecessarily "fixing" what we would be doing here. eg: the library could well be implemented independently and not be a part of the `pypa` organization here.

Reply:
@pradyunsg you wrote:

> opinion: It is important to separate the discussion of the standard from the discussion about the implementation.
> 
> This section (and the discussion of a library being developed for licensing) should probably be dropped from the PEP -- it's not relevant to the metadata update and going into too much detail of how-to-do-this, unnecessarily "fixing" what we would be doing here. eg: the library could well be implemented independently and not be a part of the `pypa` organization here.

This makes sense, though I added the section about a reference implementation based on a comment from @pfmoore

FWIW, the library already exists and not part of Pypa alright.

Reply:
I went back and forth and ended keeping this as a reworded reference implementation section for now. That said, I am quite happy to remove that section if there is a consensus that it does not belong there.